src/plugins/unified_search/public/saved_query_form/save_query_form.tsx

```diff
@@ -64,23 +64,16 @@ export function SaveQueryForm({
     }
   );
 
-  useEffect(() => {
-    const fetchQueries = async () => {
-      const allSavedQueries = await savedQueryService.getAllSavedQueries();
-      const sortedAllSavedQueries = sortBy(allSavedQueries, 'attributes.title');
-      setSavedQueries(sortedAllSavedQueries);
-    };
-    fetchQueries();
-  }, [savedQueryService]);
-
-  const validate = useCallback(() => {
+  const validate = useCallback(async () => {
     const errors = [];
-    if (
-      !!savedQueries.find(
-        (existingSavedQuery) => !savedQuery && existingSavedQuery.attributes.title === title
-      )
-    ) {
-      errors.push(titleConflictErrorText);
+
+    if (!savedQuery) {
+      const { queries }: { queries: Array<SavedQuery | undefined> } =
+        await savedQueryService.findSavedQueries(title, 1, 1);
+
+      if (queries[0]?.attributes.title === title) {
+        errors.push(titleConflictErrorText);
```

Comment:
I was able to create saved queries with the same name via API. Should we add this check on backend too?